src/psyclone/parse/module_manager.py

```diff
@@ -39,12 +39,21 @@
 
 from collections import OrderedDict
 import copy
+from difflib import SequenceMatcher
 import os
+import re
 
 from psyclone.errors import InternalError
+from psyclone.parse.file_info import FileInfo
 from psyclone.parse.module_info import ModuleInfo
 
 
+# regex to find Fortran module names. Have to be careful not to match
+# e.g. "module procedure :: some_sub".
+_MODULE_PATTERN = re.compile(r"^\s*module\s+([a-z]\S*)\s*$",
```

Comment:
Could this be a class variable initialised with None, and in `get` only compile it? While afaik python will only execute the `compile` the first time this module is imported, I don't think there is a need  to expose this symbol unnecessary.

Reply:
Oh yes. In fact I've just made it an instance variable since this is a Singleton anyway.